airflow/contrib/hooks/bigquery_hook.py

```diff
@@ -45,20 +47,20 @@ class BigQueryHook(GoogleCloudBaseHook, DbApiHook):
     Interact with BigQuery. This hook uses the Google Cloud Platform
     connection.
     """
-    conn_name_attr = 'bigquery_conn_id'
+    conn_name_attr = 'bigquery_conn_id'  # type: str
 
     def __init__(self,
-                 bigquery_conn_id='google_cloud_default',
-                 delegate_to=None,
-                 use_legacy_sql=True,
-                 location=None):
+                 bigquery_conn_id: str = 'google_cloud_default',
+                 delegate_to: str = None,
+                 use_legacy_sql: bool = True,
+                 location: str = None) -> None:
         super().__init__(
             gcp_conn_id=bigquery_conn_id, delegate_to=delegate_to)
         self.use_legacy_sql = use_legacy_sql
         self.location = location
         self.num_retries = self._get_field('num_retries', 5)
 
-    def get_conn(self):
+    def get_conn(self) -> "BigQueryConnection":
```

Comment:
It's a real class, so everything is fine.

Reply:
Could you explain more?

Reply:
In this case it is a valid class because this hook has an unusual implementation because it integrates with the database.

Reply:
But why use this string not `BigQueryConnection`?

Reply:
It's a foward reference. Because this class is later defined in the file. In Python 3.7  you will be able to refer to classes that are defined later in annotations, but now you must use such syntax.
https://www.python.org/dev/peps/pep-0484/
https://www.python.org/dev/peps/pep-0563/

Reply:
I got it but still... can't we move this class to the top of the file and use normal annotation? Of course this is my subjective feeling that this `"my_class_type"` is quite ugly.

Reply:
We can try to do it, but in a separate PR to make the review easier. 

On the other hand, it's ugly, but that's how Python works. In the future, Python's behavior will change and then the problem will disappear. Probably a person who works much more with types in Python more often sees these types of situations. Now you don't necessarily like this change, but just the current Python implementation of annotation, but...Python behavioral changes are not covered by our current SOW 😸

Reply:
It will be solve in Python 4.0. For Python 3.7+ due to [PEP-563 - Postponed Evaluation of Annotations](https://www.python.org/dev/peps/pep-0563/) it is possible to use class name without string by adding `from __future__ import annotations`

Reply:
In this case I think it should stay as it is.